pkg/kv/mock_transactional_sender.go

```diff
@@ -323,3 +325,27 @@ func (f MockTxnSenderFactory) LeafTransactionalSender(tis *roachpb.LeafTxnInputS
 func (f MockTxnSenderFactory) NonTransactionalSender() Sender {
 	return f.nonTxnSenderFunc
 }
+
+// MakeMockNoSenderTxnWithTimestamp returns a txn backed by a mock sender which
+// will fail all sender operations, meaning this mock txn is container usable
+// for reading its timestamp and little else.
+func MakeMockNoSenderTxnWithTimestamp(
+	ctx context.Context,
+	actx log.AmbientContext,
+	clock *hlc.Clock,
+	gatewayNodeID roachpb.NodeID,
+	dbctx DBContext,
+	ts hlc.Timestamp,
+) (*Txn, error) {
+	db := NewDBWithContext(actx, MakeMockTxnSenderFactory(func(
```

Comment:
I know (after reading the code) that `MakeMockTxnSenderFactory` are not much of a mock per say, and it just allows you to inject Send function that does whatever you want... I am concerned however, just how easy it is to make changes to a file called `mock_transaction_sender` and inadvertently introduce unwanted dependencies (such as testing, or testify or whatnot)

I think small code duplication to implement sender and txn factory might be safer/better.

Reply:
I actually dropped mock change and just use a real txn from NewTxn now; I added the mock to prove we weren't mis-using the txn for anything other than just being a timestamp container, but now that I've proved that to myself via deleting the txn entirely in #114574 I no longer have any qualms about just doing NewTxn here for now.